animateTransform rotate now supports cx xy

ewilligers opened the pull request:
rotate transforms now support 'angle cx cy' as well as 'angle'.

http://www.w3.org/TR/SVG/animate.html#AnimateTransformElementTypeAttribute
  For a type="rotate", each value is expressed as <rotate-angle> [<cx> <cy>]

ericwilligers committed:
animateTransform rotate now supports cx xy

rotate transforms now support 'angle cx cy' as well as 'angle'.

http://www.w3.org/TR/SVG/animate.html#AnimateTransformElementTypeAttribute
  For a type="rotate", each value is expressed as <rotate-angle> [<cx> <cy>]

alancutter commented on web-animations.js:
```diff
@@ -3611,7 +3614,10 @@ function transformRE(name, numParms, hasOptionalParm) {
-  if (hasOptionalParm) {
+  if (hasOptionalParm === 2) {
+    tokenList.push(optional([SPACES, UNIT_NUMBER].join('')));
+    tokenList.push(optional([SPACES, UNIT_NUMBER].join('')));
```
SPACES matches on '' when you want to require a space between parameters. You need to use MANDATORY_SPACE (or similar name) here.

alancutter commented on web-animations.js:
```diff
@@ -3611,7 +3614,10 @@ function transformRE(name, numParms, hasOptionalParm) {
   }
   tokenList.push(UNIT_NUMBER);
-  if (hasOptionalParm) {
+  if (hasOptionalParm === 2) {
```
Booleans are not numbers! Change hasOptionalParm to numOptionalParms.

alancutter commented on web-animations.js:
```diff
@@ -3611,7 +3614,10 @@ function transformRE(name, numParms, hasOptionalParm) {
+    tokenList.push(optional([SPACES, UNIT_NUMBER].join('')));
+    tokenList.push(optional([SPACES, UNIT_NUMBER].join('')));
+  } else if (hasOptionalParm) {
     tokenList.push(optional([COMMA, UNIT_NUMBER].join('')));
```
Choosing to delimitate on commas vs spaces based on the number of optional parameters is semantically obscure, please add a parameter that specifies whether commas or spaces are used.

alancutter commented on web-animations.js:
```diff
@@ -3585,6 +3585,9 @@ var extractValues = function(values, numValues, hasOptionalValue,
   }
+  if (hasOptionalValue === 2 && values[3 + 2 * numValues]) {
+    result.push(extractValue(values, 3 + 2 * numValues, hasUnits));
+  }
```
s/hasOptionalValue/numOptionalValues/.
This should be a for loop.

alancutter commented on web-animations.js:
```diff
@@ -3585,6 +3585,9 @@ var extractValues = function(values, numValues, hasOptionalValue,
   if (hasOptionalValue && values[1 + 2 * numValues]) {
     result.push(extractValue(values, 1 + 2 * numValues, hasUnits));
```
The 1 is too magical, it should be stored in a variable that indicates what it corresponds to, in this case it appears to represent the function name.

ericwilligers committed:
Configurable optional parameters and delimiter

transformRE now allows any number of optional parameters, and accepts a
'commaDelimited' bool to indicate if the parameters should be delimited
by a comma or spaces.
